incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/config/OpenapiCompositeBindingAdapter.java

```diff
@@ -0,0 +1,798 @@
+import io.aklivity.zilla.runtime.catalog.inline.config.InlineOptionsConfig;
+import io.aklivity.zilla.runtime.catalog.inline.config.InlineSchemaConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.BindingConfig;
+import io.aklivity.zilla.runtime.engine.config.BindingConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.CompositeBindingAdapterSpi;
+import io.aklivity.zilla.runtime.engine.config.GuardedConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.ModelConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfigBuilder;
+import io.aklivity.zilla.runtime.engine.config.RouteConfigBuilder;
+import io.aklivity.zilla.runtime.guard.jwt.config.JwtOptionsConfig;
+import io.aklivity.zilla.runtime.model.core.config.IntegerModelConfig;
+import io.aklivity.zilla.runtime.model.core.config.StringModelConfig;
+import io.aklivity.zilla.runtime.model.json.config.JsonModelConfig;
+import io.aklivity.zilla.runtime.vault.filesystem.config.FileSystemOptionsConfig;
+
+public final class OpenapiCompositeBindingAdapter implements CompositeBindingAdapterSpi
+{
+    private static final String INLINE_CATALOG_NAME = "catalog0";
+    private static final String INLINE_CATALOG_TYPE = "inline";
+    private static final String VERSION_LATEST = "latest";
+    protected static final Pattern JSON_CONTENT_TYPE = Pattern.compile("^application/(?:.+\\+)?json$");
+
+    private final Matcher jsonContentType = JSON_CONTENT_TYPE.matcher("");
+    private final Map<String, ModelConfig> models = Map.of(
+        "string", StringModelConfig.builder().build(),
+        "integer", IntegerModelConfig.builder().build()
+    );
+
+    private OpenApi openApi;
+    private int[] allPorts;
+    private int[] httpPorts;
+    private int[] httpsPorts;
+    private boolean isPlainEnabled;
+    private boolean isTlsEnabled;
+    private Map<String, String> securitySchemes;
+    private boolean isJwtEnabled;
+
+    @Override
+    public String type()
+    {
+        return OpenapiBinding.NAME;
+    }
+
+    public OpenapiCompositeBindingAdapter()
+    {
+
+    }
+
+    @Override
+    public BindingConfig adapt(
+        BindingConfig binding)
+    {
+        OpenapiOptionsConfig options = (OpenapiOptionsConfig) binding.options;
+        OpenapiConfig openapiConfig = options.openapis.get(0);
+        this.openApi = openapiConfig.openapi;
+
+        this.allPorts = resolveAllPorts();
+        this.httpPorts = resolvePortsForScheme("http");
+        this.httpsPorts = resolvePortsForScheme("https");
+        this.isPlainEnabled = httpPorts != null;
+        this.isTlsEnabled = httpsPorts != null;
+        this.securitySchemes = resolveSecuritySchemes();
+        this.isJwtEnabled = !securitySchemes.isEmpty();
+
+        switch (binding.kind)
+        {
+        case CLIENT:
+            return resolveClientBinding(binding);
+        case SERVER:
+            return resolveServerBinding(binding);
+        default:
+            return binding;
+        }
```

Comment:
Let's refactor this into `OpenapiClientBindingAdapter` and `OpenapiServerBindingAdapter` with common super class `OpenapiBindingAdapter` and abstract method `.adapt(BindingConfig)`, then store in an enum map from `binding.kind` to `UnaryFunction<BindingConfig>`, with default `Function.identity()` which will return the same `binding` unchanged.